GUIDELINES.md

```diff
@@ -1,3 +1,694 @@
+
+```c#
+/// <summary>  
+///  This is the class definition of CustomClass.  
+/// </summary>  
+public class CustomClass{} 
+```
+A list of recommended tags for documentation comments can be found in [Microsoft C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/recommended-tags-for-documentation-comments).
+
+Visual Studio produces compiler-generated XML documents by taking the XML comments from the code and building them into an XML file. This XML file can then be used to generate human-readable documentation in a variety of forms including HTML, MSDN style documentation and Intellisense within the code window.
+
+Our team uses [Monodoc](http://http://www.mono-project.com/docs/tools+libraries/tools/monodoc/) as a tool to build HTML and Intellisense documentation based on the compiler-generated XML files.
+
+Recommended elements to document:
+
+* Class description
+* Property
+* Enum type
+* Constant
+* Method
+
+
+## Brace Style
+
+All braces get their own line as it is a C# convention:
+
+**For example:**
+
+```c#
+class CustomClass
+{
+    void DoSomething()
+    {
+        if (someTest)
+        {
+            // ...
+        }
+        else
+        {
+            // ...
+        }
+    }
+    
+    void DoSomethingElse(int n)
+    {
+    	for (int i = 0; i < n; ++i) 
+    	{
+		    // ...
+		}
+    }
+}
+```
+
+**Not:**
+
+```c#
+class CustomClass {
+    void DoSomething() {
+        if (someTest) {
+          // ...
+        } else {
+          // ...
+        }
+    }
+    void DoSomethingElse(int n)
+    {
+    	for (int i = 0; i < n; ++i) {
+		    // ...
+		}
+    }
+}
+```
+
+
+## New Line
+
+Blank lines improve readability. They set off blocks of code which are in themselves logically related.
+
+Two lines should always be used between:
+
+* Logical sections of a source file
+* Class and interface definitions
+
+One line should always be used between:
+
+*  Methods
+*  Properties
+*  Local variables in a method and its first statement
+*  Logical sections inside a method to improve readability
+
+Note that blank lines must be indented as they would contain a statement this makes insertion in these lines much easier.
+
+## Wrapping Lines
+
+When an expression will not fit on a single line, break it up according to these general principles:
+
+* Break after a comma
+* Break after an operator
+* Prefer higher-level breaks to lower-level breaks
+* Indentation for line wraps should use 4 spaces
+
+### Long Argument Lists
+
+When your argument list grows too long, split your method invocation across multiple lines, with the first argument on a new line after the opening parenthesis of the method invocation, the closing parenthesis of the invocation on its own line at the same indentation level as the line with the opening parenthesis.
```

Comment:
When your argument list grows too long, split your method invocation across multiple lines. With the first argument on a new line after the opening parenthesis of the method invocation, and the closing parenthesis of the invocation on its own line at the same indentation level as the line with the opening parenthesis.